java/adapter/avro/src/test/java/org/apache/arrow/AvroToArrowTest.java

```diff
@@ -122,6 +124,66 @@ public void testNestedRecordType() throws Exception {
     checkNestedRecordResult(schema, data, root);
   }
 
+  @Test
+  public void testSkipFields() throws Exception {
+    Set<String> skipFieldNames = new HashSet<>();
+    skipFieldNames.add("f1");
+    config = new AvroToArrowConfigBuilder(config.getAllocator()).setSkipFieldNames(skipFieldNames).build();
+    Schema schema = getSchema("test_record.avsc");
+    Schema expectedSchema = getSchema("test_skip_fields.avsc");
+
+    ArrayList<GenericRecord> data = new ArrayList<>();
+    ArrayList<GenericRecord> expectedData = new ArrayList<>();
+
+    for (int i = 0; i < 5; i++) {
+      GenericRecord record = new GenericData.Record(schema);
+      record.put(0, "test" + i);
+      record.put(1, i);
+      record.put(2, i % 2 == 0);
+      data.add(record);
+
+      GenericRecord expectedRecord = new GenericData.Record(expectedSchema);
+      expectedRecord.put(0, record.get(0));
+      expectedRecord.put(1, record.get(2));
+      expectedData.add(expectedRecord);
+    }
+
+    VectorSchemaRoot root = writeAndRead(schema, data);
+    checkRecordResult(expectedSchema, expectedData, root);
+  }
+
+  @Test
+  public void testSkipNestedFields() throws Exception {
+    Set<String> skipFieldNames = new HashSet<>();
+    skipFieldNames.add("f0.f0");
```

Comment:
this only covers f0.f0, i'd like to see this test updated to have a third nesting level.